mlx_graphs/utils/convert.py

```diff
@@ -34,15 +37,63 @@ def to_networkx(
 
     G = nx.DiGraph()
 
+    if data.graph_features:
+        G.graph["graph_features"] = data.graph_features.tolist()
+
+    if data.graph_labels:
+        G.graph["graph_labels"] = data.graph_labels.tolist()
+
     if data.num_nodes is None:
         return G
 
-    G.add_nodes_from(range(data.num_nodes))
+    # G.add_nodes_from(range(data.num_nodes))
+
+    node_features_present = data.node_features is not None
+    node_labels_present = data.node_labels is not None
 
-    for v, w in data.edge_index.T.tolist():
+    node_attrs = {}
+    for i in range(data.num_nodes):
+        if node_features_present:
+            node_attrs["features"] = data.node_features[i].tolist()
+        if node_labels_present:
+            node_attrs["label"] = data.node_labels[i].item()
+        G.add_node(i, **node_attrs)
+
+    edge_features_present = data.edge_features is not None
+    edge_attrs = {}
+    for i, (v, w) in enumerate(data.edge_index.T.tolist()):
         if remove_self_loops and v == w:
             continue
-
-        G.add_edge(v, w)
+        if edge_features_present:
+            edge_attrs["features"] = data.edge_features[i].tolist()
+        G.add_edge(v, w, **edge_attrs)
 
     return G
+
+
+def from_networkx(data: Any) -> GraphData:
+    data_dict = defaultdict(list)
+
```

Comment:
You may need to import networkx in here

Reply:
What we usually do is to provide a message to the user if the package is not installed (see the try/except block [here](mlx_graphs/datasets/lanl_dataset.py))

Reply:
I think I need to import networkx before as we want a hard dependency on input
`from_networkx` needs to have `nx.Graph` as input and that requires networkx to be imported before

Reply:
Yes, you can do it in a try/except in the imports in the preamble.